sherlock-platform/customization/src/com/google/sherlock/onboarding/EmptyProjectGenerator.java

```diff
@@ -0,0 +1,121 @@
+/***
+ * Copyright 2024 Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***/
+
+package com.google.sherlock.onboarding;
+
+import com.intellij.facet.ui.ValidationResult;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.platform.DirectoryProjectGenerator;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import static org.jetbrains.jps.TimingLog.LOG;
+
+public class EmptyProjectGenerator implements DirectoryProjectGenerator<NewProjectAction.MyProjectSettings> {
+  @Override
+  public @Nullable @Nls(capitalization = Nls.Capitalization.Sentence) String getDescription() {
+    return null;
+  }
+
+  @Override
+  public @Nullable String getHelpId() {
+    return null;
+  }
+
+
+  @Override
+  public @NotNull String getName() {
+    return "Sherlock";
+  }
+
+  @Override
+  public @Nullable Icon getLogo() {
+    return null;
+  }
+
+  @Override
+  public void generateProject(@NotNull Project project, @NotNull VirtualFile baseDir,
+                              @NotNull NewProjectAction.MyProjectSettings settings, @NotNull Module module) {
+    ProgressManager.getInstance().run(new Task.Modal(project, "Creating My Project", true) {
+      @Override
+      public void run(@NotNull ProgressIndicator indicator) {
+        indicator.setText("Creating project directory...");
+        createProjectDirectory(baseDir, settings);
+        indicator.setText("Adding sample files...");
+        addSampleFiles(baseDir, settings);
+        //TODO:  add project generation logic here eventually
+      }
+    });
+  }
+
+
+  @Override
+  public @NotNull ValidationResult validate(@NotNull String baseDirPath) {
+    Path baseDir = Paths.get(baseDirPath);
+    if (!Files.exists(baseDir)) {
```

Comment:
Should we invert this logic? If there is an existing directory, we probably want to reject it. We want Sherlock to be in charge of creating the directory (and all the files inside that directory).

Then we can remove the "expects an empty directory" section below.